cmake/modules/SearchInstalledSoftware.cmake

```
@@ -1402,6 +1402,11 @@ if(vdt OR builtin_vdt)
 endif()
 
 #---Check for VecGeom--------------------------------------------------------------------
+if(vecgeom AND builtin_veccore)
+  message(WARNING "VecGeom doesn't support external VecCore (veccore & builtin_veccore options), but only native VecCore built together with VecGeom. Option VecGeom will be disabled.")
+  set(vecgeom OFF CACHE BOOL "Disabled because VecGeom doesn't support external Vecore" FORCE)
```

I don't understand the message. I would understand "external" as opposed to "builtin" - but that's not what this means, right?
I known the term "native" only from "native build or cross-compilation", but again that's probably not what's meant here. Given the `if`, let me just guess what could be a possible reworded version:

```suggestion
  message(WARNING "ROOT must be built against the VecCore installation that was used to build VecGeom; builtin_veccore cannot be used. Option VecGeom will be disabled.")
  set(vecgeom OFF CACHE BOOL "Disabled because non-builtin VecGeom specified but its VecCore cannot be found" FORCE)
```

---

I would argue that this should be an error: if I specify `-Dvecgeom=On` and then CMake turns it off - then *at least* for `fail-on-missing` this should error out?

---

Ok, I will add it in fail-on-missing